spec/models/manageiq/providers/openstack/cloud_manager/template_spec.rb

```diff
@@ -0,0 +1,27 @@
+describe ManageIQ::Providers::Openstack::CloudManager::Template do
+  let(:ems) { FactoryGirl.create(:ems_openstack_with_authentication) }
+  let(:image_attributes) { {:name => "image", :ram => "1"} }
+  let(:template_openstack) { FactoryGirl.create :template_openstack, :ext_management_system => ems }
+  let(:service) { double }
+
+  context 'when raw_delete_image' do
+    before do
+      allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems)
+      allow(ems).to receive(:with_provider_connection).with(:service => 'Compute').and_yield(service)
+    end
+
+    subject { template_openstack }
+
+    it 'should delete image' do
+      expect(service).to receive(:delete_image).once
```

Comment:
PR looks good last few things about test. It would be good to test if the ```service``` received method ```delete_image``` with correct (subject's)  _ems_ref_.

Could you update the test accroding to it (and potentially delete not used variables from let(:image_attributes))?

Reply:
Updated

Reply:
Sorry, still missing ems_ref in delete_image call..